appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/HealthCheckService.java

```diff
@@ -116,16 +115,11 @@
     private ApplicationRegistry applicationRegistry;
 
     @Inject
-    private MetricsHealthCheckConfiguration configuration;
-
-    @Inject
-    private ConfigProviderResolver microConfigResolver;
-
-    private boolean backwardCompEnabled;
+    private MicroprofileHealthCheckConfiguration configuration;
 
     private static final Logger LOG = Logger.getLogger(HealthCheckService.class.getName());
 
-    private final Map<String, Set<HealthCheck>> health = new ConcurrentHashMap<>();
+    private final Map<String, Set<HealthCheck>> health = unmodifiableMap(new HashMap<>());
```

Comment:
Can't wrap my head around this one. How is an empty modifiable map of any use? Why would a concurrent map be wrong in a shared service? How is this only applying to `health` and not equally to `readiness` and `liveness`?
Maybe you can shine some light on this for me.

Reply:
I did it in the interest of "non invasive changes" whilst still trying to make it clear that this isn't modified anymore. Perhaps it would be more readable if I just use an explicit new HashMap each time instead. I'll refactor this